text/0000-deployment-language.md

```diff
@@ -0,0 +1,561 @@
+end;
+
+# define a flow - connecting artefact instances
+define flow my_flow
+links
+    # reference artefacts by id - these instances are not recreated if they are already created
+    connect "/connector/my_http_connector/01:out" to "/pipeline/passthrough:in";
+    # implicit default ports - here we auto-create instances
+    connect passthrough to my_file;
+end;
+
+# instantiate a flow - let the events flow ~~~
+create instance "01" from flow/my_flow;
+```
+
+
+
+Explain the proposal as if it was already included in Tremor and you were teaching it to another Tremor user. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how stakeholders should *think* about the feature, and how it should impact the way they use tremor. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing tremor stakeholders and new tremor programmers.
+
+For implementation-oriented RFCs (e.g. for language internals), this section should focus on how language contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Troy supports 2 very basic operations / kinds of statements:
+
+ * Definition of artefacts
+ * Creation of artefacts
+
+## Artefact definition
+
+Artefact definition creates a referencable artefact inside the tremor repository.
+It contains:
+
+ * an id that is unique per
+ * intrinsic connector type this artefact is based on
+ * generic connector config
+ * specific config for the connector type
+ * possibly a list of arguments that can/need to be provided upon creating/instantiating this artefact
+
+There are many artefacts like connectors, sources, sinks, pipelines, deployments.
+Here we want to describe 3 special artefacts: Connectors, Pipelines, Deployments:
+
+
+### Connectors
+
+EBNF Grammar:
+
+```
+connector       = "define" connector_type "connector" artefact_id
+                  [ "args" argument_list ]
+                  ( with_block | script_block )
+                  "end"
+connector_type  = [ module_path "::" ] id      
+module_path     = id [ "::" module_path ]
+artefact_id     = id     
+argument_list   = ( id | id "=" expr ) [ "," argument_list ]
+with_block      = "with" assignment_list
+assignment_list = assignment [ "," assignment_list ]
+assignment      = let_path "=" expr
+let_path        = id [ "." id ]
+script_block    = "script" tremor_script
+```
+
+Whitespace and newlines are not significant here.
+
+**Examples:**
+
+
+
+Every connector can be configured with a script block that returns a record value. This gives maximal flexibility.
```

Comment:
I would leave this out of an initial RFC and wait until we have a need for this. It's always easier to add funciotnality then to take it away and the simpler a solution is, the less alternatives exist the easeir tis is to learn.